VideoLocker/src/main/java/org/edx/mobile/authentication/LoginTask.java

```diff
@@ -48,9 +50,33 @@ public AuthResponse call() throws Exception {
 
         // get profile of this user
         if (response.isSuccess()) {
-            response.profile = api.getProfile();
+            try {
+                response.profile = api.getProfile();
+            } catch (IOException e) {
+                throw new RetroHttpException(RetrofitError.networkError(null, e));
+            } catch (Exception e) {
+                /* TODO: Move the API for getting the user profile data
+                 * to Retrofit, and then post a message appropriate for
+                 * the type of error encountered during the request.
+                 */
+                throwLoginException();
+            }
+            if (!response.hasValidProfile()) {
+                throwLoginException();
+            }
 
         }
         return response;
     }
+
+    /**
+     * Throw a generic exception about login failure.
+     *
+     * @throws LoginException The exception.
```

Comment:
Is **The exception** necessary here ?
It'd be better if we use `{@link LoginException}`.